inference-engine/thirdparty/movidius/XLink/shared/src/XLinkDispatcher.c

```diff
@@ -361,19 +393,41 @@ int DispatcherWaitEventComplete(xLinkDeviceHandle_t *deviceHandle)
         return -1;
     }
 
-    int rc = XLink_sem_wait(id);
+    int rc = 0;
+    if (timeoutMs != XLINK_NO_RW_TIMEOUT) {
+        // This is a workaround for sem_timedwait being influenced by the system clock change.
+        // This is a temporary solution. TODO: replace this with something more efficient.
+        while (timeoutMs--) {
+            rc = XLink_sem_trywait(id);
+            if (!rc) {
+                break;
+            } else {
+#if (defined(_WIN32) || defined(_WIN64) )
+                Sleep(1);
+#else
+                usleep(1000);
+#endif
+            }
+        }
+    } else {
+        while(((rc = XLink_sem_wait(id)) == -1) && errno == EINTR)
+            continue;
+    }
```

Comment:
Hi @Maxim-Doronin @DariaMityagina 
Is this change Windows related? If so, the `sem_timedwait` calls this ported `clock_gettime` function 
```
int clock_gettime(int dummy, struct timespec *spec)
{
    dummy;	// for unreferenced formal parameter warning
    __int64 wintime; GetSystemTimeAsFileTime((FILETIME*)&wintime);
    wintime -= 116444736000000000LL;  //1jan1601 to 1jan1970
    spec->tv_sec = wintime / 10000000LL;           //seconds
    spec->tv_nsec = wintime % 10000000LL * 100;      //nano-seconds
    return 0;
}
```

Which seems to retrieve system clock instead of monotonic (also `win_time.h` defines `CLOCK_MONOTONIC` as 0, so if a check is added, that should be changed as well)